### DISCUSSION TITLE
Merge snakeyaml-engine-kmp into kaml

### DISCUSSION THREAD

**Discussion by krzema12** [charleskorn/kaml#557] [General]
Hi @charleskorn! [snakeyaml-engine-kmp](https://github.com/krzema12/snakeyaml-engine-kmp) folks here 😄 
(CC @aSemy)

We'd like to discuss the idea of moving snakeyaml-engine-kmp to become a Gradle module in kaml, think: kaml-engine. We realized that in practice, one would rarely want to use snakeyaml-engine-kmp directly, and use it through kaml anyway. It would also be easier to deploy fixes if both live in the same repo. I recall you were open to discussing such option, so now when the libraries are integrated and work fine, we thought it's a good time to revisit.

Other thoughts/remarks:

* Together with @aSemy we'd still contribute to the project, in particular: sync it periodically with the upstream snakeyaml-engine, but also introduce original improvements, like making the library fully compliant with the YAML test suite (@aSemy has some WIP here). It would be a nice feature for kaml, a good differentiator
* It would be good to publish this module as well from within kaml, just in case someone wants to use the event-based API
* We'd take care of moving the project to kaml repo, so for you it would be mostly about reviewing the PRs
* The only significant thing would be the git history. It'd be nice to preserve it, but it would require merging unrelated histories. Which is possible of course, but we'd want to take care not to lose the git history in a squash merge (because git history is sensitive to file location changes)

Please let us know what you think!

**Comment by charleskorn** [comment]
Thanks for starting this discussion @krzema12!

Unfortunately I am finding myself with less and less time to contribute to kaml and this is unlikely to change any time soon. So I'm hesitant to add more code to the otherwise relatively stable codebase, as I won't be able to dedicate much time to PR reviews etc. 

> We realized that in practice, one would rarely want to use snakeyaml-engine-kmp directly

I'm not so sure about this: there are plenty of scenarios where using SnakeYAML directly makes sense instead of using kaml, and I see the same scenarios applying to snakeyaml-engine-kmp too.

**Comment by krzema12** [comment]
@charleskorn 
Ok, got it! We can keep the projects as separate ones. Anyway, together with @aSemy we're willing to help with maintaining the project.
